arroyo/processing/strategies/run_task_with_multiprocessing.py

```diff
@@ -483,24 +488,29 @@ def __check_for_results_impl(self, timeout: Optional[float] = None) -> None:
                 self.__next_step.submit(message)
 
             except MessageRejected:
-                result.next_index_to_process = idx
-                next_step_has_applied_backpressure = True
-                break
+                # For the next invocation of __check_for_results, start at this message
+                result.valid_messages_transformed.reset_iterator(idx)
 
-        if result.next_index_to_process != len(input_batch):
-            if next_step_has_applied_backpressure:
                 self.__metrics.increment(
                     "arroyo.strategies.run_task_with_multiprocessing.batch.backpressure"
                 )
-            else:
-                self.__metrics.increment(
-                    "arroyo.strategies.run_task_with_multiprocessing.batch.output.overflow"
+
+                # This is a warning because backpressure on the multiprocessing
+                # strategy _might_ cause degraded CPU saturation.
+                logger.warning(
+                    "Received backpressure from next_step, splitting output batch (%0.2f%% complete)",
+                    idx / len(input_batch) * 100,
                 )
```

Comment:
What do you think about omitting this warning entirely? IMO it's a bit too noisy and the backpressure metric on the stream processor seems like it's enough. None of the other strategies emit something similar to this.

Reply:
ok makes sense, removed